### PR TITLE
Implement Netuitive Reporter

### PR DESCRIPTION
**Pull request by joepusateri**
Initial implementation of Netuitive Dropwizard Reporter to StatsD

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,69 +1,66 @@
...
+```
+This directory contains a `docker-compose.yml` file which contains a test stack definition for testing.
+In that directory run: 
+
+```
+docker-compose up -d --build
+```
+Now create and run your tests
+
+`./run_tests.sh`
```
Comment (TheConnMan) [inline README.md:67]: Where are these tests?

File: README.md [hunk 2]
```diff
@@ -1,69 +1,66 @@
...
+
+### Running the tests
+```
+cd src/test/docker;
+```
+This directory contains a `docker-compose.yml` file which contains a test stack definition for testing.
+In that directory run: 
+
+```
+docker-compose up -d --build
```
Comment (TheConnMan) [inline README.md:63]: This fails after a clean `git clone`. The dropwizard example application does a COPY of a .jar which is expected to be prebuilt. The Dockerfile should build the example app within the container.

File: src/main/java/io/dropwizard/metrics/NetuitiveMetricsReporter.java [hunk 3]
```diff
@@ -139,14 +140,143 @@ public NetuitiveMetricsReporter build(final String ananke) {
...
+	 * @param metricName
+	 *            Base metric name for this Meter
+	 * @param meter
+	 *            Metered object to report
+	 */
+    void pushMeter(String metricName, Metered meter)
+    {
+    	pushGauge(metricName + ".count", new Double(meter.getCount()));
+    	pushGauge(metricName + ".meanRate", meter.getMeanRate());
+    	pushGauge(metricName + ".1MinuteRate", meter.getOneMinuteRate());
```
Comment (TheConnMan) [inline src/main/java/io/dropwizard/metrics/NetuitiveMetricsReporter.java:221]: Should metric names have any caps in them? I tend to see only lower case metric FQNs.

File: src/test/docker/docker-compose.yml [hunk 4]
```diff
@@ -3,6 +3,11 @@ services:
   dropwizard:
     build: ./docker-dropwizard-example/
     container_name: dropwizard
+    image: example/dropwizard
```
Comment (TheConnMan) [inline src/test/docker/docker-compose.yml]: Only one of build and image should be used in a `docker-compose.yml` file, so this image line shouldn't be here.

File: src/test/docker/docker-dropwizard-example/README.md [hunk 5]
```diff
@@ -1,49 +1,77 @@
...
-WORKDIR /opt/dropwizard
-CMD ["java", "-jar", "-Done-jar.silent=true", "docker-dropwizard-application-standalone.jar", "server", "config.yml"]
+# Docker Dropwizard image
+
+Example Docker image for running a Dropwizard Application in a container.
+
+Requires:
+* [Docker](https://www.docker.com/)
+* [Boot2Docker](http://boot2docker.io/)
+* JDK (to compile java file locally)
```
Comment (TheConnMan) [inline src/test/docker/docker-dropwizard-example/README.md]: Requiring a JDK and Gradle for a local Docker build is against best practice.

File: src/test/docker/docker-dropwizard-example/dockerBuildImage.bat [hunk 6]
```diff
@@ -0,0 +1 @@
+docker build -t example/dropwizard:latest .
```
Comment (TheConnMan) [inline src/test/docker/docker-dropwizard-example/dockerBuildImage.bat]: Why are helper bat/sh necessary? These should just be in the docs (as it looks like they are).

**Review by TheConnMan — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (joepusateri) [reply README.md:63]: I had not checked in the built jars.  Instructions in the docker example project shows how to build via gradle.  This readme is about using the library in the user's project.  More Readme changes expected via Jordan

File: README.md (hunk 1 above)
Comment (joepusateri) [reply README.md:67]: User to provide their own tests

File: src/main/java/io/dropwizard/metrics/NetuitiveMetricsReporter.java (hunk 3 above)
Comment (joepusateri) [reply src/main/java/io/dropwizard/metrics/NetuitiveMetricsReporter.java:221]: These follow other tools Reporter implementation names

File: src/test/docker/docker-compose.yml (hunk 4 above)
Comment (joepusateri) [reply src/test/docker/docker-compose.yml]: Using an image here names it in the local repo.  Doing this matches the image name given in the dockerBuildImage script.

File: src/test/docker/docker-dropwizard-example/README.md (hunk 5 above)
Comment (joepusateri) [reply src/test/docker/docker-dropwizard-example/README.md]: I don't know what to do with this

File: src/test/docker/docker-dropwizard-example/dockerBuildImage.bat (hunk 6 above)
Comment (joepusateri) [reply src/test/docker/docker-dropwizard-example/dockerBuildImage.bat]: They are not necessary, but I found them useful.

File: src/test/docker/docker-dropwizard-example/README.md (hunk 5 above)
Comment (TheConnMan) [reply src/test/docker/docker-dropwizard-example/README.md]: The Zorka agent Dockerfile is an example of building all necessary artifacts within the container: https://github.com/Netuitive/zorka/blob/master/Dockerfile

File: src/test/docker/docker-compose.yml (hunk 4 above)
Comment (TheConnMan) [reply src/test/docker/docker-compose.yml]: I see, it's for labeling the image, not using the image. Good to know.

File: src/test/docker/docker-dropwizard-example/dockerBuildImage.bat (hunk 6 above)
Comment (TheConnMan) [reply src/test/docker/docker-dropwizard-example/dockerBuildImage.bat]: Feels like repo clutter to have 8 bat/sh scripts in the repo just to run one line java and docker commands. I understand they're helpful locally, but I don't think they should be checked in.

File: README.md (hunk 1 above)
Comment (TheConnMan) [reply README.md:67]: Will any metrics come into Netuitive from Dropwizard without user written tests?

File: README.md (hunk 2 above)
Comment (TheConnMan) [reply README.md:63]: https://github.com/Netuitive/dropwizard-metrics/pull/1#discussion_r100550875 is the more relevant discussion for this, let's continue there.

File: README.md (hunk 1 above)
Comment (joepusateri) [reply README.md:67]: TBH I am not sure what these tests would consist of.  Once started the dropwizard example will report metrics once per minute with mostly zero values.

File: src/test/docker/docker-dropwizard-example/dockerBuildImage.bat (hunk 6 above)
Comment (joepusateri) [reply src/test/docker/docker-dropwizard-example/dockerBuildImage.bat]: I will remove them

**Review by TheConnMan — APPROVED** [review]

**Pull request merged**